pyro/infer/abstract_infer.py

```diff
@@ -56,14 +56,16 @@ def _dist_and_values(self, *args, **kwargs):
                 logits.append(logit)
             else:
                 # Value has already been seen.
-                logits[ix] = util.log_sum_exp(torch.stack([logits[ix], logit]).squeeze())
+                logits[ix] = util.log_sum_exp(torch.stack([logits[ix], logit]))
 
         logits = torch.stack(logits).squeeze()
         logits -= util.log_sum_exp(logits)
         if not isinstance(logits, torch.autograd.Variable):
             logits = Variable(logits)
         logits = logits - util.log_sum_exp(logits)
-
+        # Categorical distribution must have at least one event dimension
```

Comment:
Would it be cleaner to replace the `.squeeze()` above with a `.view()`?
```diff
- logits = torch.stack(logits).squeeze()
+ logits = torch.stack(logits).contiguous().view(-1)
```

Reply:
Done